### DISCUSSION TITLE
Copy & Paste: JTextField and JTextArea

### DISCUSSION THREAD

**Discussion by APOC10** [coobbi/Jre4android#9] [Q&A] [answered]
Hello.  I have a number of simple Swing UI programs which have JTextfield(s) for input of strings and JTextAreas for outputing calculations / derivations from the strings.

I have found that using DefaultEditorKit.CopyAction() and new DefaultEditorKit.PasteAction does not function in Jre4Android in my programs.  I can paste in text into the JTextField by using the Android GBoard keyboard, but I cannot copy text from the JTextArea either by using DefaultEditorKit.CopyAction() and then pasting it into a ,txt file, or by using the Android GBoard to copy and paste.

I am unable to save the contents of the JTextArea to a .txt file as well, which closes off that option.

Is there a way to copy text from a JTextArea and paste it into an app like HoverNote or some other text editor?  I can select text, and cut it, but when I paste it, the result is just a blank line.

Any help would be immensely appreciated, thank you.

**Comment by coobbi** [comment]
@APOC10 
Can you provide a simple demo jar file to me via email: imzine.com@gmail.com.  I will use this jar to see if I can implement this function.

**Reply by APOC10** [reply, marked as answer]
Thanks for answering.  I have sent a simple example program to the email you provided.  I hope this gets to you OK.  Thanks.

**Comment by coobbi** [comment]
New version can be download from google play @APOC10
